src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -37,6 +39,57 @@
  */
 public abstract class QueryBuilder<QB extends QueryBuilder> extends ToXContentToBytes implements Writeable<QB> {
 
+    public static enum Operator {
+        OR, AND;
+
+        public BooleanClause.Occur toMustOrShouldClause() {
+            switch (this) {
+                case OR:
+                    return BooleanClause.Occur.SHOULD;
+                case AND:
+                    return BooleanClause.Occur.MUST;
+                default:
+                    throw Operator.newOperatorException(this.toString());
+            }
+        }
+
+        public static Operator readFrom(StreamInput in) throws IOException {
+            switch (in.readVInt()) {
+                case 0:
+                    return OR;
+                case 1:
+                    return AND;
+                default:
+                    throw new ElasticsearchException("Serialized operator needs to be either '1' or '0'");
```

Comment:
Also, this applied to all error messages, we tend to use square brackets instead of `'` around values. Also messages are usually lowercased (just a non written convention)